graphql-language/src/ast.rs

```diff
@@ -7,22 +7,23 @@ use source::Source;
 
 /// Contains some character offsets that identify where the source of the AST
 /// is from.
+#[derive(Clone, Copy)]
 pub struct Location<'a> {
   pub start: usize,
   pub end: usize,
-  pub source: Option<&'a Source<'a>>
+  pub source: Source<'a>
 }
 
 /// All AST node types implement this trait.
 pub trait Node<'a> {
-  fn location(&self) -> Option<&Location<'a>>;
```

Comment:
But what happens when code later down the line depends on a `location`? Also, what about AST transformation and generation? That's a non-test case that could be harmed by a dummy location. For example a directive might need to alter the AST in some way, there would be no `Location` available in that case.